protocol/strict-derivation.md

```diff
@@ -0,0 +1,269 @@
+
+Those checks are done once per span batch before the batch queue derives any singular batches.
+
+# Strict Batch Ordering
+
+Strict batch ordering requires batches within and across channels to be strictly ordered.
+This removes the necessity to buffer more than one channel, and the channel bank effectively holds
+at most one staging channel, for which it collects frames, that must arrive in order.
+
+The following will explore the open design space, presenting options, and propose a solution.
+The principle I would apply to answer open design questions is that the new rules should generally
+choose the design that leads to faster progess of derivation, even if it means deriving
+empty batches earlier.
+Spoiler: Option 1 will always be the proposed option.
+
+## Out of order frames
+
+There's an open design space around how to handle some scenarios for missing or out of order frames:
+- How do we handle incoming _first_ frames while the current channel isn't closed yet?
+  - Option 1: discard the current channel and open a new one for the given frame.
+  - Option 2: discard the foreign frame and wait for the duration of the channel timeout for the
+    next correct frame and only then discard the current staging channel and open a new staging
+    channel for the next first frame.
+- How do we handle the current staging channel if an out of order frame (that is not the first
+frame) arrives? There's only one valid option, to drop the out-of-order frame, keeping the currently
+staged channel. Note that this will give the staging channel a chance to be continued or closed
+properly by the next in-order frame. If this doesn't happen, at the lastest the next first frame
+will drop the staging channel and start a new one.
+
+### Proposed solution
+
+Applied to above open question, this means choosing option 1. This has the
+additional advantage that the sync start algorithm simplifies significantly, because a fist frame
+_always_ marks the start of a new channel, and guarantees that there won't be any buffered channels
+in the channel bank.
+
+## Out of order batches
+
+The batch queue also becomes simpler as batches have to arrive in order. 
+
+- How to treat _older_ batches is easy: they are dropped.
+- Similarly to out of order frames, how to treat future batches (gaps) has two options:
+  - Option 1: immediately fill the gap with deposit-only blocks, and then process the batch.
+  - Option 2: drop future batches, and wait for up to a full sequencing window for the next
+  batch to arrive.
+
+### Proposed solution
+
+Following the principle of fast derivation, option 1 is proposed to be chosen. Besides guaranteeing
+faster derivation, it also avoids waiting for full sequencing windows on gaps to be resolved, which
+is quite unlikely, unless the batcher implementation would be specially tuned to backfill gaps. If a
+gap occurs, it is actually much more likely that the next batches will be even further in the
```

Comment:
Maybe some very unlikely of an L1 reorg, that doesn't cause an L2 reorg yet because the unsafe chain uses older L1 origins, then that L1 reorg removes the lastest batcher tx and the next channel is included by the txmgr automatically adjusting the nonce. Very constructed case, but not impossible.